src/index.js

```diff
@@ -39,18 +45,70 @@ const parseFromStd = (stdout, stderr) => {
 const getProjectDirectory = filePath =>
                               atom.project.relativizePath(filePath)[0] || path.dirname(filePath);
 
+
+// Retrieves style guide documentation with cached responses
+const getMarkDown = async (url) => {
+  const anchor = url.split('#').slice(-1);
+  const docsExipired = (lastRetrieved, lifetime) => new Date().getTime() - lastRetrieved < lifetime;
+
+  if (docsRuleCache[anchor] && docsExipired(docsLastRetrieved, DOCUMENTATION_LIFETIME)) {
+    return docsRuleCache[anchor];
+  }
+
+  let rawRulesMarkdown;
+  try {
+    rawRulesMarkdown = await get('https://raw.githubusercontent.com/bbatsov/ruby-style-guide/master/README.md');
+  } catch (x) {
+    return '***\nError retrieving documentation';
+  }
+
+  const byLine = rawRulesMarkdown.split('\n');
+  const ruleAnchors = byLine.filter(l => l.match(/\* <a name=/g));
+
+  docsRuleCache = ruleAnchors.reduce((cache, rambo) => {
+    // not efficient but itll do
+    const startingLine = byLine.indexOf(rambo);
+    const ruleName = rambo.split('"')[1];
+    const beginSearch = byLine.slice(startingLine + 1);
+
+    // gobble all the documentation until you reach the next rule
+    const documentationForRule = takeWhile(beginSearch, x => !x.match(/\* <a name=/));
+    const markdownOutput = '***\n'.concat(documentationForRule.join('\n'));
+
+    return { [ruleName]: markdownOutput, ...cache };
+  }, docsRuleCache);
+
+  docsLastRetrieved = new Date().getTime();
+  return docsRuleCache[anchor];
+};
+
 const forwardRubocopToLinter =
-  ({ message: rawMessage, location, severity, cop_name }, filePath) => {
-    const [message, url] = rawMessage.split(/ \((.*)\)/, 2);
+  ({ message: rawMessage, location, severity, cop_name: copName }, filePath) => {
+    const [excerpt, url] = rawMessage.split(/ \((.*)\)/, 2);
     const { line, column, length } = location;
-    const outputToLinter = {
-      type: ['refactor', 'convention', 'warning'].includes(severity) ? 'Warning' : 'Error',
-      html: formatMessage({ cop_name, message, url }),
-      filePath,
+
+    const severityMapping = {
+      refactor: 'info',
+      convention: 'info',
+      warning: 'warning',
+      error: 'error',
+      fatal: 'error',
     };
-    return Object.assign(outputToLinter,
-      location && { range: [[line - 1, column - 1], [line - 1, (column + length) - 1]] },
-    );
+
+    const linterMessage = {
+      url,
+      excerpt: `${excerpt} (${copName})`,
+      severity: severityMapping[severity],
+      description: url ? getMarkDown.bind(this, url) : null,
+    };
+
+    if (location) {
```

Comment:
`location` is required for v2 messages! If you aren't given one from `rubocop` you're going to have to make one up (`helpers.generateRange(textEditor, 0)` to highlight the first line of the file).